### PR TITLE
RUL-39: add converters for dates and metrics

### PR DESCRIPTION
**Pull request by mmetayer**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Component/Product/Updater/Converter/DateToStringDataConverter.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convert(ValueInterface $sourceValue, AttributeInterface $targetAttribute)
+    {
+        Assert::isInstanceOf($sourceValue->getData(), \DateTimeInterface::class);
+
+        return $sourceValue->getData()->format('c');
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Component/Product/Updater/Converter/DateToStringDataConverter.php]: Are you sure for the `c` format? The users fill a date with day month and year. Not sure it's important to then to have the hour, minute second and the timezone. For the concatenate, the default format is `Y-m-d` (US date format) but they could specify the format in the settings.  
To check with Morgane.

File: tests/back/Pim/Enrichment/Specification/Component/Product/Updater/Converter/MetricToStringDataConverterSpec.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    function it_converts_a_metric_to_a_string()
+    {
+        $sourceValue = MetricValue::value(
+            'weight_in_grams',
+            new Metric('weight', 'GRAM', 23.55, 'KILOGRAM', .02355)
+        );
+
+        $this->convert($sourceValue, new Attribute())->shouldReturn('23.5500 GRAM');
```
Comment (nmarniesse) [inline tests/back/Pim/Enrichment/Specification/Component/Product/Updater/Converter/MetricToStringDataConverterSpec.php]: Number format must be validate with Morgane. If I remember well, we store `23.5500` in DB but users see `23,55` in UI. Seem to be more logic for users to have the second one when they copy (that's we did for concatenate)

File: src/Akeneo/Pim/Enrichment/Component/Product/Updater/Converter/DateToStringDataConverter.php (hunk 1 above)
Comment (mmetayer) [reply src/Akeneo/Pim/Enrichment/Component/Product/Updater/Converter/DateToStringDataConverter.php]: That's the expected format, as stated in https://akeneo.atlassian.net/browse/RUL-39

File: tests/back/Pim/Enrichment/Specification/Component/Product/Updater/Converter/MetricToStringDataConverterSpec.php (hunk 2 above)
Comment (mmetayer) [reply tests/back/Pim/Enrichment/Specification/Component/Product/Updater/Converter/MetricToStringDataConverterSpec.php]: The point of this class is to copy attribute values, not transforming them. IMO the output should be as close as possible as the input.

Comment (mmetayer) [reply tests/back/Pim/Enrichment/Specification/Component/Product/Updater/Converter/MetricToStringDataConverterSpec.php]: Seen with @morganen: keep it like that for now, and we can iterate later if needed

**Pull request merged**